mpags-cipher.cpp

```diff
@@ -0,0 +1,244 @@
+// mpags-cipher.cpp
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+int main( int argc, char* argv[]){
+
+    const std::vector<std::string> cmdLineArgs{argv, argv+argc};
+
+    // Creating a boolean variable to tell us when help us requested.
+    bool helpRequested{false};
+
+    // Creating a boolean variable to tell me when the version is requested.
+    bool versionRequested{false};
+
+    // Creating a boolean to check if the input file was given and initialising the input file variable
+    bool input_fileGiven{false};
+    std::string input_file{""};
+
+    // Creating a boolean to check if the output file was given and initialising the output file variable
+    bool output_fileGiven{false};
+    std::string output_file{""};
+
+    // Looking through the command line arguments, excluding the first one.
+    for (std::size_t i{1}; i < cmdLineArgs.size(); i++){
```

Comment:
Since `cmdLineArgs` is const, it is arguably better to store the result of `cmdLineArgs.size()` in a variable and compare with that here, to avoid calling the function at each iteration of the loop.  The actual impact on timing is likely to be negligible in this case but if you have a loop that will execute millions or even billions of times, it might start to add up.